tools/unicode_case_conversion.py

```diff
@@ -0,0 +1,628 @@
+
+    table_description = """/* Provides information of how many one-to-one, one-to-two or one-to-three mapping
+ * added successively into the `upper_case_conversions` table.
+ */"""
+    data.append(create_c_format_table('uint8_t', 'upper_case_conversion_counters', upper_case_conversions[1], table_description))
+
+    with open(c_source, 'w') as genereted_source:
+        genereted_source.write(''.join(data))
+
+
+class GenTables(object):
+    """Class defines an informative, default generated tables."""
+
+    def __init__(self, lower_case_table, upper_case_table):
+        """
+        Generate the extracted tables from the given case mapping tables.
+
+        :param lower_case_table: Lower-case mappings.
+        :param upper_case_table: Upper-case mappings.
+        """
+
+        self._character_case_ranges = lower_case_table.extract_ranges(upper_case_table)
+        self._character_pair_ranges = lower_case_table.extract_character_pair_ranges(upper_case_table)
+        self._character_pairs = lower_case_table.extract_character_pairs(upper_case_table)
+        self._upper_case_special_ranges = upper_case_table.extract_special_ranges()
+        self._lower_case_ranges = lower_case_table.extract_ranges()
+        self._lower_case_conversions = lower_case_table.extract_conversions()
+        self._upper_case_conversions = upper_case_table.extract_conversions()
+
+        if lower_case_table.get_table():
+            warnings.warn('Not all elements extracted from the lowercase conversion table!')
+        if upper_case_table.get_table():
+            warnings.warn('Not all elements extracted from the uppercase conversion table!')
+
+    def get_character_case_ranges(self):
+        return self._character_case_ranges
+
+    def get_character_pair_ranges(self):
+        return self._character_pair_ranges
+
+    def get_character_pairs(self):
+        return self._character_pairs
+
+    def get_upper_case_special_ranges(self):
+        return self._upper_case_special_ranges
+
+    def get_lower_case_ranges(self):
+        return self._lower_case_ranges
+
+    def get_lower_case_conversions(self):
+        return self._lower_case_conversions
+
+    def get_upper_case_conversions(self):
+        return self._upper_case_conversions
+
+
+def main():
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument('--unicode-data', metavar='FILE', action='store', required=True, help='specify the unicode data file')
+    parser.add_argument('--special-casing', metavar='FILE', action='store', required=True, help='specify the special casing file')
+    parser.add_argument('--c-source', metavar='FILE', action='store', default=C_SOURCE_FILE, help='specify the output c source (default: %(default)s)')
+
+    script_args = parser.parse_args()
+
+    if not os.path.isfile(script_args.unicode_data) or not os.access(script_args.unicode_data, os.R_OK):
+        print ('The %s file is missing or not readable!' % script_args.unicode_data)
```

Comment:
I don't think we need spaces before the `(`. No other place has a space in that place in this file.